log.hh

```diff
@@ -46,4 +49,18 @@ inline void apply_settings(const settings& s) {
 using seastar::pretty_type_name;
 using seastar::level_name;
 
+} // namespace logging
+
+template <typename... Args>
+sstring log_format(logger& l, log_level level, const char* fmt, Args... args) {
+    if (l.is_enabled(level)) {
+        return format(fmt, std::forward<Args>(args)...);
+    } else {
+        return "";
+    }
+}
+
+template <typename... Args>
+sstring debug_format(logger& l, const char* fmt, Args... args) {
+    return log_format(l, log_level::debug, fmt, std::forward<Args>(args)...);
```

Comment:
Not sure I understand the motivation. Also, see bc75e2c1d182ae8440333b50feb71bb183dbfe57

Reply:
The motivation is to conditionally format a message string only when the logger's debug log level is enabled.
The message string is prepared in several call sites and then passed to a common function that logs it, again only in debug level.

An alternative, if we want to normally save these cycles could be to pass a function that will generate the message and call it only if debug log level is enabled, or do the debug-logging at the caller, before calling the common function (`mutate_token_metadata`)